modules/file/file.go

```diff
@@ -1478,6 +1516,72 @@ func (r *run) checkHash(f fileEntry, hashtype checkType) {
 	return
 }
 
+func (r *run) checkBytes(f fileEntry) {
+	var (
+		err error
+	)
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("checkBytes() -> %v", e)
+		}
+	}()
+	reader := f.getReader()
+	defer f.Close()
+
+	// Must use the "label,search" for loop iteration because of
+	// map[string]search for parameters
+	for _, search := range r.Parameters.Searches {
+		if !search.isactive {
+			continue
+		}
+
+		for _, c := range search.checks {
+
+			decodedByte := make([]byte, hex.DecodedLen(len(c.bytes)))
+			_, err := hex.Decode(decodedByte, c.bytes)
+			if err != nil {
+				e := fmt.Sprintf("unable to decode byte query: %v ", err)
+				panic(e)
+			}
+			scansize := len(decodedByte) // Get the actual byte slice length
+			blocksize := 4 * (2 << 10)   // 4096, 4k
+
+			var bigbuf = make([]byte, (blocksize + (2 * scansize)))
+			// Read slightly more than 4k so we have a full buffer to scan through
+			// as we move into the for loop
+			_, err = reader.Read(bigbuf)
+			if err != nil && err != io.EOF {
+				panic(err)
+			}
+
+			for err != io.EOF {
+				if bytes.Contains(bigbuf, decodedByte) {
+					// We do find the intended hex values in our scan
+					// Break the loop and return
+					// Not certain exactly how to do that here
+					c.storeMatch(f.filename)
+					break
+				} else { //Explicitly, we do not find the hex value, and have not encountered EOF
+					tempBuf := make([]byte, blocksize)
+					_, err = reader.Read(tempBuf)
```

Comment:
Off hand, I'm not sure if this part is correct. As an example, in this loop we will keep reading the file until we encounter EOF. The problem is we are within the loop iterating over the checks already.

So, say for example we have two bytes checks in the same action. The second will never be used to match anything since we will have read the entire file while scanning for the first check.

In the standard regex content checking function, the file reads occur outside the checks loop and each check is then applied to the read in buffer.

e.g., https://github.com/mozilla/mig/blob/35ea4f9f2f2a9bb42d7df83b9afdf55d4b544b75/modules/file/file.go#L1263-L1281